tests/validate/ps_iops_test/src/ps_iops.cpp

```diff
@@ -0,0 +1,241 @@
+/**
+* Copyright (C) 2022 Xilinx, Inc
+*
+* Licensed under the Apache License, Version 2.0 (the "License"). You may
+* not use this file except in compliance with the License. A copy of the
+* License is located at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations
+* under the License.
+*/
+#include <iostream>
+#include <fstream>
+#include <iomanip>
+#include <vector>
+#include <chrono>
+#include <thread>
+
+#include "cmdlineparser.h"
+
+#include "xilutil.hpp"
+#include "xrt/xrt_device.h"
+#include "xrt/xrt_bo.h"
+#include "xrt/xrt_kernel.h"
+
+using ms_t = std::chrono::microseconds;
+using Clock = std::chrono::high_resolution_clock;
+
+static const int COUNT = 1024;
+static const size_t DATA_SIZE = COUNT * sizeof(int);
+
+typedef struct task_args {
+  int thread_id;
+  int queueLength;
+  unsigned int total;
+  Clock::time_point start;
+  Clock::time_point end;
+} arg_t;
+
+struct krnl_info {
+    std::string     name;
+    bool            new_style;
+};
+
+bool verbose = false;
+barrier barrier;
+struct krnl_info krnl = {"hello_world", false};
+
+static void usage(const char *prog)
+{
+  std::cout << "Usage: " << prog << " <Platform Test Area Path> [options]\n"
+    << "options:\n"
+    << "    -d       device index\n"
+    << "    -t       number of threads\n"
+    << "    -l       length of queue (send how many commands without waiting)\n"
+    << "    -a       total amount of commands per thread\n"
+    << "    -v       verbose result\n"
+    << std::endl;
+}
+
+double runTest(std::vector<xrt::run>& cmds, unsigned int total, arg_t &arg)
+{
+  int i = 0;
+  unsigned int issued = 0, completed = 0;
+  arg.start = Clock::now();
+
+  for (auto& cmd : cmds) {
+    cmd.start();
+    if (++issued == total)
+      break;
+  }
+
+  while (completed < total) {
+    cmds[i].wait();
+
+    completed++;
+    if (issued < total) {
+      cmds[i].start();
+      issued++;
+    }
+
+    if (++i == cmds.size())
```

Comment:
It is typically not a great idea to place commands that have effects into conditionals. Not sure what our standards are on them. It would be better to just write
```
i++;
if (i == cmds.size())
  i = 0;
```